### ISSUE TITLE
Use the --rules fast path for severity-only per-file config groups

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1185]
## Context

Umi switched `umi lint` to the `@utoo/lint` `CLIEngine` compatibility API in [umijs/umi#13426](https://github.com/umijs/umi/pull/13426), currently using `@utoo/lint@0.3.4`.

The native CLI already has an efficient rule-selection path through `--rules`. The npm wrapper also emits this flag for compatible non-materialized configs.

However, [`nativeConfigRunsForFiles()`](https://github.com/utooland/utoo-lint/blob/10c9821e859c4f6901299072380e0528a5f64617/npm/utoo-lint/index.js#L3173-L3232) currently converts every configured per-file group into a temporary config containing all native rules disabled plus the resolved rule map, and sets `forceMaterializedConfig: true`. That explicitly bypasses the existing [`--rules` selection path](https://github.com/utooland/utoo-lint/blob/10c9821e859c4f6901299072380e0528a5f64617/npm/utoo-lint/index.js#L3346-L3362).

This happens even when the resolved group contains only rule severities and has no rule options or shared settings. It adds configuration serialization/parsing work to the common Umi case and makes the compatibility API materially slower than invoking the native linter with the same enabled rules.

## Expected behavior

When a resolved per-file group contains only pure severity values and no settings, run that group with an equivalent command such as:

```sh
utoo-lint --no-config --rules=rule-a,rule-b <paths...>
```

The wrapper can continue applying the resolved warning/error severities while converting native diagnostics to ESLint-compatible results. Config groups containing rule options or settings should keep using a materialized config.

## Acceptance criteria

- [ ] Detect severity-only rule maps, including string, numeric, boolean, and one-item array severity forms.
- [ ] Invoke the native binary through `--no-config --rules=...` for eligible per-file groups without generating an all-disabled temporary rule map.
- [ ] Preserve `off`, warning, and error behavior, including `CLIEngine` result severities, exit status, and fix behavior.
- [ ] Fall back to the materialized-config path when any enabled rule has options or the group has supported shared settings.
- [ ] Add tests that distinguish the fast path from the materialized fallback.
- [ ] Add a regression benchmark using a large severity-only config representative of the Umi integration.

## Relationship to other work

This optimizes each group produced by the current Node-side file resolution. Moving `files` and `ignores` matching into the native layer is a separate, larger optimization.
